packages/aws-cdk-lib/aws-servicecatalog/lib/private/product-stack-synthesizer.ts

```diff
@@ -10,45 +9,42 @@ import { ProductStack } from '../product-stack';
  * Interoperates with the StackSynthesizer of the parent stack.
  */
 export class ProductStackSynthesizer extends cdk.StackSynthesizer {
-  private readonly assetBucket?: IBucket;
   private bucketDeployment?: BucketDeployment;
+  private parentAssetBucket?: IBucket;
 
-  constructor(assetBucket?: IBucket) {
+  constructor(private readonly parentDeployment: cdk.IStackSynthesizer, private readonly assetBucket?: IBucket) {
```

Comment:
This okay in principle, but we tend to prefer props objects.
Because I have just approved an other PR that also adds stuff to this: https://github.com/aws/aws-cdk/pull/26303/files#diff-26862b8f0be4955057aed48a0d3dfabefaa27d96e95d3c5c99a3a85cc10ce770R18

Can you rebase your PR and then turn this whole thing into a prop object please?